src/core/session.ts

```diff
@@ -133,9 +133,25 @@ export class Session implements FormSubmitObserverDelegate, HistoryDelegate, Lin
 
   followedLinkToLocation(link: Element, location: URL) {
     const action = this.getActionForLink(link)
-    this.visit(location.href, { action })
+    this.convertLinkWithMethodClickToFormSubmission(link) || this.visit(location.href, { action })
   }
 
+  convertLinkWithMethodClickToFormSubmission(link: Element) {
+    const linkMethod = link.getAttribute("data-turbo-method")
+
+    if (linkMethod) {
+      const form = document.createElement("form")
+      form.method = linkMethod
```

Comment:
So, this only supports data-turbo-method of get and post, not delete?

Reply:
It does work with any method that works in a form AFAIK, are you encountering any issue?

Reply:
My mistake.  When initially testing, it didn't work correctly.    Since It creates the form differently in Rails UJS (post with hidden delete input), I assumed it didn't work.  Digging into it more, it works as expected.

Reply:
Actually, this is a potential issue with the Rails stack.  Prior UJS library converted links to POST and added hidden _method.  This made handling redirect_to easy.  Now, if using DELETE or POST, Turbo XHR redirects do not work unless adding a status: 3xx on the response.  (see #320).

Reply:
Interesting, I think rails `form_for` and other form helpers actually use a `POST` form and a `_method` attribute. 

But I think `_method` is a rails construct right ? So it would not work with another backend, which is not what we want :(